syntax = "proto3";

package api;

import "google/protobuf/timestamp.proto";
import "google/api/annotations.proto";
import "google/protobuf/empty.proto";
import "protoc-gen-openapiv2/options/annotations.proto";
import "api/prompter.proto";

service Predictor {
  rpc PrepareData(PrepareDataReq) returns (google.protobuf.Empty) {};
  rpc Predict(PredictReq) returns (PredictResp) {};
  rpc UniqueCodes(google.protobuf.Empty) returns (UniqueCodesResp) {};
}

message PrepareDataReq {
  repeated string sources = 1;
}

message PredictReq {
  QueryType type = 1;
  string product = 2;
  string period = 3;
}

message PredictResp {
  bytes data = 1;
}

message UniqueCodesResp {
  repeated UniqueCode codes = 1;
}

message UniqueCode {
  string segment = 1;
  string name = 2;
  bool regular = 3;
}